src/Emulator.ts

```diff
@@ -118,14 +134,24 @@ implements IGetGenesisInfos, IGetProtocolParameters, IResolveUTxOs, ISubmitTx
                     this.pushUtxo(new UTxO({
                         resolved: tx.body.outputs[i],
                         utxoRef: new TxOutRef({
-                            id: new Hash32('a'.repeat(64)),
+                            id: tx.hash.toString(), // to be fixed
                             index: i
                         })
                     }))
                 }
+                
             }
+            // to do - update this.slot and this.time
+            // to do - Move time forward to the next block number
+            // to do - Update the block number
```

Comment:
We should add a blockNumber or height in the Emulator class that we can increment by the height